v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/cassandra/iowrapper/CassandraConnector.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.cassandra.iowrapper;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.CqlSessionBuilder;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.ProgrammaticDriverConfigLoaderBuilder;
+import com.datastax.oss.driver.api.core.session.Session;
+import com.google.cloud.teleport.v2.source.reader.io.cassandra.schema.CassandraSchemaReference;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import org.jline.utils.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Autocloseable Connector For Cassandra. Note: This class is not Serializable as it holds an active
+ * connection. {@link org.apache.beam.sdk.transforms.PTransform Ptransform} must store {@link
+ * CassandraDataSource} instead and use the connector to open a connection. TODO(vardhan):
+ * Optionally allow driver config from GCS file.
+ */
+public final class CassandraConnector implements AutoCloseable {
+  private final CqlSession session;
+  private static final Logger LOG = LoggerFactory.getLogger(CassandraConnector.class);
+
+  public CassandraConnector(
+      CassandraDataSource dataSource, CassandraSchemaReference schemaReference) {
+    Log.info(
+        "Connecting to Cassandra Source dataSource = {}, schemaReference = {}",
```

Comment:
Just verifying, but this prints the actual object and not the pointer right?

Reply:
That was the case for `java.sqlx.DataSOurce` as it was a very opaque object.  `CassandraDataSource` is built using `AutoValue` so it comes by default for string pretty printing.

The jdbc issue got fixed when we introduced an extension to basic java.sqlx.DataSource for Jdbc.